gcovr/html_generator.py

```diff
@@ -332,9 +375,10 @@ def print_html_report(covdata, output_file, options):
         os.chdir(options.root_dir)
         with io.open(data['filename'], 'r', encoding=options.source_encoding,
                      errors='replace') as source_file:
+            data['lexer_used'] = formatter.set_lexer(data['filename'])
             for ctr, line in enumerate(source_file, 1):
                 data['source_lines'].append(
-                    source_row(ctr, line.rstrip(), cdata.lines.get(ctr))
+                    source_row(ctr, formatter.format_line(line), cdata.lines.get(ctr))
```

Comment:
I think (but am not sure) that Pygments should be used to highlight the entire file contents in one go. We can split the lines afterwards (Pygment's HTML formatter makes sure that no elements cross a line). This would matter for C++ raw strings or preprocessor directives with line continuation.

Reply:
You're right, I've fotgot multi-line definitions.